File: Audio Filtering, the FFT, and Doppler Shifts/AudioLab/AudioLab/TableViewController.m
//
//  TableViewController.m
//  AudioLab
//
//  Created by Dhaval Gogri on 9/20/18.
//  Copyright © 2018 Eric Larson. All rights reserved.
//

#import "TableViewController.h"
#import "ViewController.h"
#import "ViewControllerModuleB.h"

@interface TableViewController () <UITableViewDelegate, UITableViewDataSource>

@end

@implementation TableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.tableView.delegate = self;
    self.tableView.dataSource = self;
}


- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

// has 2 rows in the table for 2 modules.
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return 2;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    if(indexPath.row == 0){
            ViewController *viewController = [self.storyboard instantiateViewControllerWithIdentifier:@"ViewControllerModuleA"];
            [self.navigationController pushViewController:viewController animated:YES];
    }
    else if(indexPath.row == 1){
        NSLog(@"View DID Load 1");
        ViewControllerModuleB *viewControllerB = [self.storyboard instantiateViewControllerWithIdentifier:@"ViewControllerModuleB"];
        [self.navigationController pushViewController:viewControllerB animated:YES];
        NSLog(@"View DID Load 2");
    }
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"TableViewCell" forIndexPath:indexPath];
    if(indexPath.row == 0){
        cell.textLabel.text = @"Module A";
    }
    else if(indexPath.row == 1){
        cell.textLabel.text = @"Module B";
    }
    return cell;
}


@end

File: Core Motion and SpriteKit/Lab3/Lab3/TableViewController.swift
//
//  TableViewController.swift
//  Lab3
//
//  Created by Dhaval Gogri on 9/29/18.
//  Copyright © 2018 Dhaval Gogri. All rights reserved.
//

import UIKit

// This class only shows Module for Lab 3
class TableViewController: UITableViewController {

    @IBOutlet var tableViewModule: UITableView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.tableViewModule.delegate = self;
        self.tableViewModule.dataSource = self;
    }

    // MARK: - Table view data source

    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 3
    }

    // Shows 3 options - Module A, B and Settings
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "tableViewCell", for: indexPath)
        
        if(indexPath.row == 0){
            cell.textLabel?.text = "Module A - Pedometer";
        }
        else if(indexPath.row == 1){
            cell.textLabel?.text = "Module B - Game";
        }
        else if(indexPath.row == 2){
            cell.textLabel?.text = "Module - Settings";
        }
        
        return cell
    }
    
    // Navigation on cell click to a particular Module
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        if(indexPath.row == 0){
            let storyBoard: UIStoryboard = UIStoryboard(name: "Main", bundle: nil)
            let viewController = storyBoard.instantiateViewController(withIdentifier: "PedoMeterStoryBoard") as! ViewController
            self.navigationController?.pushViewController(viewController, animated: true);
        }
        else if(indexPath.row == 1){
            let storyBoard: UIStoryboard = UIStoryboard(name: "Main", bundle: nil)
            let viewController = storyBoard.instantiateViewController(withIdentifier: "GameStoryBoard") as! GameViewController
            self.navigationController?.pushViewController(viewController, animated: true);
        }
        else if(indexPath.row == 2){
            let storyBoard: UIStoryboard = UIStoryboard(name: "Main", bundle: nil)
            let viewController = storyBoard.instantiateViewController(withIdentifier: "SettingsStoryBoard") as! SettingsViewController
            self.navigationController?.pushViewController(viewController, animated: true);
        }
        
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        // Keep orientation as per the screen orientation
        let appDelegate = UIApplication.shared.delegate as! AppDelegate
        appDelegate.enableAllOrientation = true
    }
    
    
}
